File: Temporal/BarrysPeanutsDotNet/Model/CreditCard.cs
namespace BarrysPeanuts;

public class CreditCard: ICreditCard
{
    public string Name { get; set; }

    public string Number { get; set; }

    public Address BillingAddress { get; set; }

    public int ExpirationMonth { get; set; }

    public int ExpirationYear { get; set; }

    public int SecurityCode { get; set; }

    public CreditCard(ICreditCard creditCard)
    {
        Name = creditCard.Name;
        Number = creditCard.Number;
        BillingAddress = creditCard.BillingAddress;
        ExpirationMonth = creditCard.ExpirationMonth;
        ExpirationYear = creditCard.ExpirationYear;
        SecurityCode = creditCard.SecurityCode;
    }
}

File: Temporal/BarrysPeanutsJava/app/src/main/java/barryspeanuts/model/CreditCard.java
package barryspeanuts.model;

public class CreditCard {
    String fullName;
    String number;
    int expirationMonth;
    int expirationYear;
    int securityCode;

    public CreditCard(){}

    public CreditCard(String fullName, String number, int expirationMonth, int expirationYear, int securityCode) {
        this.fullName = fullName;
        this.number = number;
        this.expirationMonth = expirationMonth;
        this.expirationYear = expirationYear;
        this.securityCode = securityCode;
    }
}
